src/interfaces.rs

```diff
@@ -276,21 +278,51 @@ impl Interfaces {
 
         Ok(())
     }
+
+    /// Add an interface from a json file
+    pub fn add_interface_file(&mut self, file_path: &str) -> Result<(), AstarteError> {
+        let interface = Interface::from_file(file_path.as_ref())?;
+        let name = crate::interface::traits::Interface::name(&interface);
+        debug!("Added interface {}", name);
+        self.interfaces.insert(name.to_owned(), interface);
+        Ok(())
+    }
+
+    /// Add all json interface description inside a specified directory
+    pub fn add_interface_directory(
+        &mut self,
+        interfaces_directory: &str,
+    ) -> Result<(), AstarteError> {
+        let interface_files = std::fs::read_dir(std::path::Path::new(interfaces_directory))?;
+        let it = interface_files.filter_map(Result::ok).filter(|f| {
+            if let Some(ext) = f.path().extension() {
```

Comment:
How about using `ends_with` to check extension ?
```suggestion
            .filter(|f| f.path().ends_with("json"));
```